include/odp/api/spec/classification.h

```diff
@@ -573,6 +575,9 @@ typedef enum {
 	/** Destination SCTP port (val_sz = 2) */
 	ODP_PMR_SCTP_DPORT,
 
+	/** GTPU Tunnel endpoint identifier  (val_sz = 4)*/
+	ODP_PMR_GTPU_TEID,
```

Comment:
I wonder how GTP TEID matching really works. To my understanding (but I may be wrong) the existing terms, like ODP_PMR_TCP_DPORT, match if ODP has determined that the packet is of the right protocol (TCP) and that the protocol specific matching criteria (port number matches) is satisfied.

In case of GTP, ODP cannot distinguish whether a received UDP packet is GTP or something else since the interpretation is up to the application. Thus the GTP TEID term requires better, unambiguous definition of when exactly it matches.

If the GTP TEID PMR term just looks at UDP packets at certain offset, then does it bring much value over the existing custom PMR term (though perhaps custom L4 term would be more convenient).

Reply:
We will send the next version with the definition of GTPU in the comment.